lib/include/dlisio/stream.hpp

```diff
@@ -64,6 +64,16 @@ class stream {
 
     void close() noexcept (true);
     int eof() const noexcept (true);
+
+    /* Leaf protcol
+     *
+     * Gives user access to the most underlying layer of the stream.
+     *
+     * Note that this method must be used with extreme caution as
+     * both current and returned stream would work on the same underlying
+     * object.
+     */
+    stream leaf() noexcept (false);
```

Comment:
We only need the eof state, not the entire handle. Would add `bool peof()` instead. The problem with a new stream of the underlying tell, is that if someone starts messing with it, they break the original stream. At the very least what you return here should be const.